### ISSUE TITLE
Tons of deprecated packages and documentation out of date

### ISSUE THREAD

**Issue by OSP123** [FirebaseExtended/firepad#307]
**Firebase:** "^3 || ^4"

**Firepad:** 1.5.0

**Ace:** Not using Ace

**CodeMirror:** Not using CodeMirror

**Other (e.g. Node, browser, operating system) (if applicable):** Chrome, Mac OSX

### Steps to reproduce

- Easy to reproduce by just cloning to local or pulling latest from master and attempting to run a simple `npm install`.

### Expected behavior

- Application should install all the relevant packages without issue. Also, the documentation is out of date concerning the moncao-editor integration. 

### Actual behavior

<img width="1440" alt="1" src="https://user-images.githubusercontent.com/1403723/46646557-b67fb100-cb3f-11e8-81a8-45c8ab64745e.png">
<img width="1436" alt="2" src="https://user-images.githubusercontent.com/1403723/46646558-b67fb100-cb3f-11e8-9a96-a9f757c3ad50.png">
<img width="1439" alt="3" src="https://user-images.githubusercontent.com/1403723/46646560-b67fb100-cb3f-11e8-97c7-9811f8a98b21.png">
<img width="1440" alt="4" src="https://user-images.githubusercontent.com/1403723/46646561-b67fb100-cb3f-11e8-9cfc-cf9617bc5c61.png">
<img width="1439" alt="5" src="https://user-images.githubusercontent.com/1403723/46646562-b7184780-cb3f-11e8-87f4-80f6cb68778c.png">
<img width="1438" alt="6" src="https://user-images.githubusercontent.com/1403723/46646563-b7184780-cb3f-11e8-9f74-e769e778a864.png">

Basically, deprecation and issues caused by not updating. I can try pushing up the fix for this, but not sure how much time I have.

**Comment by mikelehen** [comment]
Yep, sorry. :-(  PRs welcome!

**Comment by OSP123** [comment]
Alright, I'm pretty much done with fixing dependency issues and updating all of the packages. However, I'm still getting a weird issue with the editor not working on iOS for my deployed application using Monaco Editor.

**Comment by OSP123** [comment]
Actually, think I fixed it. Just required more updating of some of the links.

**Comment by samtstern** [comment]
@OSP123 if you want to send a PR to update the dependencies here just add me or @mikelehen as a reviewer!

**Comment by OSP123** [comment]
@samtstern Semi-PR coming your way. Thanks, man! There is a remaining integration test that's not working, but I figure you'd know more about that issue than I would. https://github.com/FirebaseExtended/firepad/pull/310

**Comment by mikelehen** [comment]
The integration test has been fixed and I added a note about monaco to the main page (not deployed yet), though I didn't add examples to the docs (sorry). For now people can look at the checked-in examples/monaco.html though.
